data_steward/analytics/cdr_ops/incremental_load_qc.py

```diff
@@ -0,0 +1,348 @@
+    ]
+    union_all_query = '\nUNION ALL\n'.join(queries)
+
+    df = execute(client, union_all_query)
+
+    success_null_check = f"No AIAN participants' records are found in {new_dataset}.<br><br>"
+    failure_null_check = (
+        "There are <b>{count}</b> tables that might have "
+        "AIAN participants' records. <br>Look at these tables and investigate why AIAN "
+        "records are there: <b>{tables}</b><br><br>")
+
+    render_message(df,
+                   success_null_check,
+                   failure_null_check,
+                   failure_msg_args={
+                       'count': len(df),
+                       'tables': ', '.join(df.table_name)
+                   })
+# -
+
+# ## QC 2. Confirm there are no duplicate `OBSERVATION_ID`s
+# Any observation records from `incremental_dataset` got new `OBSERVATION_ID`s
+# assigned in `new_dataset` to avoid ID duplicates. <br>We must confirm all the
+# `OBSERVATION_ID`s in `new_dataset` are unique. <br>The same goes to
+# `_mapping_observation` and `obsesrvation_ext`.
+
+# +
+obs_tables = ['observation', 'observation_ext'
+             ] if is_deidentified == 'True' else [
+                 'observation', '_mapping_observation', 'observation_ext'
+             ]
+
+query = JINJA_ENV.from_string('''
+    SELECT
+        \'{{table}}\' as table_name,
+        observation_id,
+        COUNT(*) AS duplicate_row_count
+    FROM `{{project}}.{{new_dataset}}.{{table}}`
+    GROUP BY table_name, observation_id
+    HAVING COUNT(*) > 1
+''')
+
+queries = [
+    query.render(project=project_id, new_dataset=new_dataset, table=table)
+    for table in obs_tables
+]
+
+union_all_query = '\nUNION ALL\n'.join(queries)
+df = execute(client, union_all_query)
+
+success_null_check = f"No OBSERVATION_ID duplicates are found in {new_dataset}.<br><br>"
+failure_null_check = (
+    "There are total <b>{count}</b> OBSERVATION_IDs that might have"
+    "duplicates. <br>Look at these tables and investigate why there are duplicates: <b>{tables}</b><br><br>"
+)
+
+render_message(df,
+               success_null_check,
+               failure_null_check,
+               failure_msg_args={
+                   'count': len(df),
+                   'tables': ', '.join(df.table_name)
+               })
+# -
+
+# ## QC 3. Confirm mapping/ext tables are consistent
+# This hotfix runs several delete and insert statements to OBSERVATION, SURVEY_CONDUCT, PERSON
+# and their mapping/ext tables. <br>We must confirm that mapping/ext tables are consistent with
+# their correspondants after the hotfix.
+
+# +
+map_ext_tuples = [
+    ('observation', 'observation_ext'),
+    ('survey_conduct', 'survey_conduct_ext'),
+    ('person', 'person_ext'),
+] if is_deidentified == 'True' else [
+    ('observation', '_mapping_observation'),
+    ('observation', 'observation_ext'),
+    ('survey_conduct', '_mapping_survey_conduct'),
+    ('survey_conduct', 'survey_conduct_ext'),
```

Comment:
So, these have the same situation as the previous `observation_ext` comment.  This will work fine for a dataset with mapping and extension tables but will fail on a dataset without both.